### DISCUSSION TITLE
Booting process with GRUB and a LUKS encrypted system

### DISCUSSION THREAD

**Discussion by RonLinu** [SpiralLinux/SpiralLinux-project#45] [General]
Hi,
I chose encryption at initial installation of Spiral Linux. Make me fell safer if my laptop is ever lost or stolen :)
I noticed that, before GRUB shows up, we are asked to enter the LUKS password. It seems GRUB is on a the same encrypted subvolume/partition  (or a separate encrypted subvolume...?).

I have Windows on a separate physical drive that is detected by Grub. Nice! But, if I want to boot it, I have to enter LUKS password, let it go through its thing before I can even access GRUB.

If GRUB is on the separate subvolume as the main OS, wouldn't it be better to leave it unencrypted and asks for LUKS password only if Linux is selected?

**Comment by ebiow** [comment]
I suppose you could use your motherboard's boot key to boot directly from your Windows drive to speed things up. That's what I do because [grub-customizer](https://easylinuxtipsproject.blogspot.com/p/grub-customizer.html) isn't detecting my Windows partition, and it just adds a huge layer of complexity to grub so I don't want to try and fix the problem manually.

My problem with the way my encrypted Spirallinux system boots is that it takes 15-30 seconds to get to where I can enter my encryption key, I guess because of the way the Calamares installer does encryption (no separate unencrypted /boot partition, which is a good thing IMO). The thing I find annoying is that if I mistype my decryption password grub takes extra long, 40 sec.-1 minute and then drops me in an EFI shell without giving me a chance to re-enter my password.

Has anyone figured out a way around these issues.

**Reply by geckolinux** [reply]
Hi there @ebiow 
> Has anyone figured out a way around these issues.

Have you observed this same behavior on other distros like Manjaro for example that have a good Calamares implementation? There are quite a few configurable options in Calamares for the distro maintainer, plus some variation between different base distros in the way they implement the initramfs and things like that. But hardly anything is configurable for LUKS as far as I can tell. That might be a good issue to raise with the [Calamares team](https://github.com/calamares/calamares/issues) since it would really depend on them to change the implementation.

**Reply by RonLinu** [reply]
Yes I know I can boot directly to Windows by going directly to the BIOS and changing boot order to speed up things. I have Windows on the internal SSD and Linux on an external SSD. 
I boot Windows only once a month so I relax :) and I boot normally despite the delay for GRUB menu to show up ("Windows boot manager" shows up in my GRUB menu).

FYI on my ASUS S550 laptop:
- 4 seconds to get to where I have to enter my encryption key
- 15 seconds to get to the GRUB menu or 32 seconds to drop to the command line if I mistyped
